tests/IlluminaSampleSheet/V1/NovaSeqSampleSheetTest.php

```diff
@@ -0,0 +1,122 @@
+<?php declare(strict_types=1);
+
+namespace MLL\Utils\Tests\IlluminaSampleSheet\V1;
+
+use MLL\Utils\IlluminaSampleSheet\V1\DataSection;
+use MLL\Utils\IlluminaSampleSheet\V1\NovaSeqHeaderSection;
+use MLL\Utils\IlluminaSampleSheet\V1\NovaSeqSampleSheet;
+use MLL\Utils\IlluminaSampleSheet\V1\ReadsSection;
+use MLL\Utils\IlluminaSampleSheet\V1\SampleSheetData;
+use MLL\Utils\IlluminaSampleSheet\V1\SettingsSection;
+use PHPUnit\Framework\TestCase;
+
+class NovaSeqSampleSheetTest extends TestCase
+{
+    public function testNovaSeqStandardSampleSheetToStringReturnsExpectedResult(): void
+    {
+        $headerSection = new NovaSeqHeaderSection(
+            '4',
+            'DonalDuck',
+            'MyExperiment',
+            '19.04.2024',
+            'MyWorkflow',
+            'MyApplication',
+            'MyAssay',
+            'MyDescription',
+            'MyChemistry',
+        );
+
+        $readsSection = new ReadsSection(101, 101);
+
+        $columns = ['Sample_ID', 'Sample_Name', 'Sample_Plate', 'Sample_Well', 'I7_Index_ID', 'Index', 'I5_Index_ID', 'Index2', 'Sample_Project', 'Description'];
+        $rows = [
+            ['1', 'Sample-001-M001', 'RunXXXX-PLATE', '', 'UDP0090', 'TCAGGCTT', 'UDP0090', 'ATCATGCG', 'RunXXXX-PROJECT', 'description'],
+            ['2',  'Sample-002-M002', 'RunXXXX-PLATE', '', 'UDP0091', 'CCTTGTAG', 'UDP0091', 'CCTTGGAA', 'RunXXXX-PROJECT', 'description'],
+            ['3', 'Sample-003-M003', 'RunXXXX-PLATE', '', 'UDP0092', 'GAACATCG', 'UDP0092', 'TCGACAAG', 'RunXXXX-PROJECT', 'description'],
+        ];
+
+        $sampleSheetData = new SampleSheetData($columns, $rows);
```

Comment:
Der Entwickler muss hier ja schon viel wissen und verschiedenste Objekte erstellen - habe da auf dem anderen Branch eine Idee angefangen - Das SampleSheet als sozusagen Aggregate Root zu sehen und die Logik wie was zusammen gebaut wird findet dann dort statt.  (https://github.com/mll-lab/php-utils/pull/22/files#diff-14076f85c01d78b993ad8d8d65102c9b41e68b8b9320c5f37c413c1ffffaeeb8)

Sieht dann in der Benutzung schon etwas einfacher aus: https://github.com/mll-lab/php-utils/pull/22/files#diff-c6ed1f992406d26529ce724db8d3b973e9376ddc48f383ac5ba9c8bb79a03c4b

(Nicht nicht ausgreift nur mal eine Idee - auch noch nicht abtrakt genug)

Reply:
ich sehe den Vorteil noch nicht ganz dieses  Aggregate Root (ich hätte es Factory genannt..). Gefühlt ist es ein level an Indirektion mehr. Anschließend muss ich asserten, welche Art von SampleSheet ich baue.
Außerdem muss ich jetzt als entwickler wissen, dass alle 4 sections existieren müssen. Die fluent setters garantieren nicht, dass alle Sectionen auch gesetzt werden, während der constructor das tut.